src/checks.rs

```diff
@@ -28,22 +29,34 @@ fn checklist() -> Vec<Box<dyn Check>> {
     ]
 }
 
+fn ending_line_checker() -> Box<dyn Check> {
+    Box::new(ending_blank_line::EndingBlankLineChecker::default())
+}
+
+fn check_and_push_warning(checker: &mut dyn Check, line: &LineEntry, warnings: &mut Vec<Warning>) {
+    if let Some(warning) = checker.run(line) {
+        warnings.push(warning);
+    }
+}
+
 pub fn run(lines: Vec<LineEntry>) -> Vec<Warning> {
     let mut checks = checklist();
     let mut warnings: Vec<Warning> = Vec::new();
 
-    for line in lines {
+    for line in &lines {
```

Comment:
If you need to check the last line separately, you can do something like this:
```rust
if let Some((last_line, rest_lines)) = lines.split_last() {
    for line in rest_lines {
        if line.is_empty_or_comment() {
            continue;
        }

        for ch in &mut checks {
            if let Some(warning) = ch.run(&line) {
                warnings.push(warning);
            }
        }
    }

    let mut ending_blank_line_check = ending_blank_line::EndingBlankLineChecker::default();
    if let Some(warning) = ending_blank_line_check.run(&last_line) {
        warnings.push(warning);
    }
}
```

Reply:
It is better, thank you.

Reply:
Done